### DISCUSSION TITLE
Position/Alt Hold

### DISCUSSION THREAD

**Discussion by ceegeester** [iNavFlight/inav#10224] [General]
I am relatively new to all this technology and being a 76 year old it sometimes becomes a bit of a challenge.  I am playing around with a basic quadcopter build and have followed lots of youtube vids to a point where I have configured the quad with basic modes angle, horizon, PosHld, AltHld and RTH. My GPS is working and I can get the thing flying. Couple of things that don't seem to work are (1) When I switch to AltHld the quad will "drop" and then sort itself out to hold Alt (2) PosHld just doesn't seem to work (3) RTH sometimes works and sometimes not.  I am using a FrySky receiver and Taranis radio with a cheapy F4V3S flight controller. Any ideas welcome.

**Comment by sensei-hacker** [comment]
> Couple of things that don't seem to work are (1) When I switch to AltHld the quad will "drop" and then sort itself out

Increase the hover throttle setting.  Right when you switch to althold, it goes to whatever you have set as hover throttle, then it adjusts from there.

For position hold and RTH, can we see how your compass and GPS is mounted. The compass / magnetometer measures the magnetic field, of course. So having the main battery wire (electro-magnet) or a buzzer near it will mess it up.

**Reply by ceegeester** [reply]
Thanks for the advice, I will check my throttle settings. Have attached a pic of the quad. I have a couple of quads one with GPS and magnetometer and one with GPS only. Is the magnetometer necessary for PosHld and RTH.
![Image00001](https://github.com/user-attachments/assets/42a59dfd-ce40-4aaa-82fc-c874b2e4ccd7)

**Comment by sensei-hacker** [comment]
A compass is recommended. Suppose the GPS indicates the quad is 1 meter south of where it should be. So it needs to move 1 meter north.  It's hard to fly 1 meter north of you don't know which direction north is.  The compass provides that critical information.

In some cases position mode can be used with the compass missing. Primarily if you continue to fly straight segments. In that case, the fact that the position is getting further and further north provides the directional reference.  That is to say - if I'm flying forward quickly and my position is getting further and further north, then forward must be north.

More information can be found in the documentation here:

https://github.com/iNavFlight/inav/wiki/GPS-and-Compass-setup

**Comment by ceegeester** [comment]
One more question while I'm on a roll. INAV multirotor settings gives me the option of setting the stick position for AltHld hover, STICK, Mid Stick or Hover. As a novice flyer which one should I choose.
